### PR TITLE
Skip flaky CommandLine.ArgumentParsing test

### PR DESCRIPTION
**Pull request by JoeRobich**

### CODE REVIEW

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:1413]: @jaredpar pretty bad to lose all the validation of this method.  should we comment out that portion taht is failing instead?

**Review by Youssef1313 — COMMENTED** [review]
I commented on the issue with a possible scenario that may be causing the failure. Instead of skipping I'd add a `catch` to the `try/finally` block, and re-throw the exception (possibly only in DEBUG). That way, we can better understand what's happening the next time it fails.

What do you think of that?

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:1413]: I also think it is better to disable specific problematic portion. Perhaps only on Linux DEBUG.

**Comment by JoeRobich** [comment]
> What do you think of that?

I believe we would certainly prefer to take a fix than skip this test.

@CyrusNajmabadi @jaredpar Thoughts about Youssef's suggestion?

**Review by CyrusNajmabadi — APPROVED** [review]

**Review by 333fred — APPROVED** [review]

**Comment by 333fred** [comment]
> I commented on the issue with a possible scenario that may be causing the failure. Instead of skipping I'd add a catch to the try/finally block, and re-throw the exception (possibly only in DEBUG). That way, we can better understand what's happening the next time it fails.

That can be done in a separate PR for investigation purposes. This test is blocking merging ability on a number of open PRs, and we need to unblock the pipeline.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

**Comment by AlekseyTs** [comment]
@JoeRobich Please include reference to the tracking issue into the commit description once you merge.

**Comment by jaredpar** [comment]
Given @Youssef1313 has a good lead on the underlying problem I would just take the most expedient step to disabling the test. Odds are we can fix and re-enable by EOD

**Review by allisonchou — APPROVED** [review]

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:7787]:
```suggestion
            #region "source"
```

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:7839]:
```suggestion
            #endregion
```

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:7841]:
```suggestion
            #region "mslib"
```

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:7906]:
```suggestion
            #endregion
```

**Comment by Youssef1313** [comment]
> Given @Youssef1313 has a good lead on the underlying problem I would just take the most expedient step to disabling the test. Odds are we can fix and re-enable by EOD

I doubt it will be by end of the day 😄 
My guess seems to be wrong. I'll try to re-investigate next week after my mid-terms in case it wasn't picked by someone from the team.

**Comment by AlekseyTs** [comment]
@JoeRobich The interesting CI jobs have passed. We probably don't want to wait another hour or so to get the integration tests green (as it stands right now, we need to re-run at least one leg). I think we should just force merge this PR, squashing the commits in the process, if possible. Note, I pushed some formatting changes to this PR to fix CI failures, those should be included.

**Comment by JoeRobich** [comment]
@AlekseyTs Squashed and merged.

**Pull request merged**
